distributed/active_memory_manager.py

```diff
@@ -657,7 +661,7 @@ def run(self) -> SuggestionGenerator:
                 # would have stopped earlier
                 continue
 
-            if len(ts.who_has) > 1:
+            if ts.who_has:
```

Comment:
this is different logic
```suggestion
            if ts.who_has and len(ts.who_has) > 1:
```